### ISSUE TITLE
Ram memory usage

### ISSUE THREAD

**Issue by benathmane** [capitalone/Hygieia#753]
Hi
We are testing and running Hygieia dashboard 2.0 on LinuxMint but we have RAM memory usage, it uses 100% of the RAM :(

![error](https://cloud.githubusercontent.com/assets/4710514/17697931/3b2d9eb8-63b8-11e6-9fa6-4f12e8b9200c.JPG)

After a while builder eats all ram memory and eventually OS kills process.

We have tried to give OS more memory via VirtualBox but no matter how much we give memory eventually it runs out.

It's normal ?! 
Thanks.

**Comment by tabladrum** [comment]
Haven't experience this ever.

**Comment by mrpudn** [comment]
It looks like you have many instances of the API and the github collector running - am I reading that correctly? Why do you need so many to be running?

**Comment by sibaom** [comment]
Ya when you are running all collector at a time then its shows memory issue though you increase the memory size.Its because of API consumption.Make sure that you are closing all connection in collector.
perform GC if you need by looking in to JConsole. In my case I increased my RAM size and have done some connection close operation and GC. now its working fine

**Comment by benathmane** [comment]
Can you execute the command line "htop" and publish the screenshot of your results to compare with my results?

**Comment by satishc1** [comment]
@benathmane 
These are numbers we have gathered while running Hygieia on an m4.xlarge AWS instance with 16Gb RAM.

Red Hat Enterprise Linux Server release 7.2 (Maipo)
Linux kernel 3.10.0-327.36.1.el7.x86_64
16GB Ram

We ran the UI module inside apache httpd. The average RAM consumption was 130 Mb for apache httpd. This will go up depending on how many users hit apache. These numbers are for light usage with about 10 people running the UI.

MongoDb 3.2.9 takes up about 1 Gb RAM.

The api process takes up about 1 Gb RAM.

We have 7 collectors running with a total RAM consumption of 7Gb. This usage will go up or down depending on the cron frequency in your properties file.
jenkins-build-collector
versionone-feature-collector
subversion-collector
github-scm-collector
aws-cloud-collector
sonar-codequality-collector
jira-feature-collector
